### PR TITLE
feat: add in new state level gen pop covid

### PR DESCRIPTION
**Pull request by nmmarquez**
New function for grabbing in state level covid data from the CDC and population data from the census so we can make state level comparisons. In the docs is an example of a plot for state level comparisons which should work for most states where we have state level prison data. Pasting below with the plot result.

```
state_ <- "Georgia"

# get state level covid data
st_gen_df <- get_genstate_covid(state_)

# get our prison data
hist_df <- alt_aggregate_counts(all_dates = T)

# get our prison denominators
pri_pop <- read_aggregate_pop_data() %>%
    filter(State == state_) %>%
    pull(Residents.Population) %>%
    first()

# calculate new cases for the state of interest
hist_df %>%
    filter(Measure == "Residents.Confirmed" & State == state_) %>%
    filter(Web.Group == "Prison") %>%
    select(Date, Residents.Confirmed = Val) %>%
    mutate(Residents.New = diff_roll_sum(Residents.Confirmed, Date)) %>%
    mutate(Residents.Population = pri_pop) %>%
    # calculate the rate using latest denominators
    mutate(Residents = Residents.New/Residents.Population) %>%
    select(Date, Residents) %>%
    # join with the overall state data
    right_join(
        st_gen_df %>%
            mutate(General.New = diff_roll_sum(General.Confirmed, Date)) %>%
            mutate(General = General.New/General.Population) %>%
            select(Date, General)) %>%
    # pivot to make plotting easier
    pivot_longer(-Date, names_to = "Group", values_to = "CR") %>%
    filter(!is.na(CR)) %>%
    mutate(Group = ifelse(
        Group == "General", state_, str_c(state_, " DOC"))) %>%
    mutate(Group = fct_rev(Group)) %>%
    filter(Date >= lubridate::ymd("2021-01-01")) %>%
    ggplot(aes(x = Date, y = CR*100000, color = Group)) +
    geom_line(size = 2) +
    theme_behindbars() +
    scale_color_bbdiscrete() +
    labs(y = "New Case Rate\nPer100,000", color = "") +
    theme(legend.position = c(.38, .92)) +
    theme(legend.background = element_rect(fill = "transparent"))

```

e60e9202-0acf-4d07-b498-d134e37f77e4

### CODE REVIEW

**Review by hjohns12 — COMMENTED** [review]
All looks good to me, and I'm sure I'll be using this often in the coming days! Do we wanna add General.VaccinesInitiated as a returned column to this function? Could do it pretty easily via code here (kudos to Erika): https://github.com/uclalawcovid19behindbars/data-viz/blob/master/ad-hoc/vaccine_blog.R#L16

**Review by erika-tyagi — APPROVED** [review]
All looks good to me too, ty!! 

Could go either way on vaccine data – might make sense as a separate function rather than a parameter in this function? Not sure though // don't have strong feelings either way!

**Comment by nmmarquez** [comment]
I think the vaccine one should be in another function unless it also has time series data but I can add that to this PR as well

**Review by hjohns12 — APPROVED** [review]

File: R/get_genstate_covid.R [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    gen_covid_df <- state_infections %>%
+        rename(state_abv = state) %>%
+        mutate(State = translate_state(state_abv)) %>%
+        mutate(State = ifelse(state_abv == "PR", "Puerto Rico", State)) %>%
+        filter(!is.na(State)) %>%
+        mutate(Date = lubridate::mdy(submission_date)) %>%
+        arrange(Date) %>%
+        select(
+            State, Date, General.Confirmed = tot_cases,
+            General.Death = tot_death) %>%
```
Comment (erika-tyagi) [inline R/get_genstate_covid.R:81]: Most minor thing, but can we call this `General.Deaths` so it's consistent with `Residents.Deaths` and `Staff.Deaths` // can use suffix shortcuts?

File: R/get_genstate_covid.R [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+#' @return a data frame with the following columns: State, Date,
+#' General.Confirmed, General.Deaths, General.Population
+#'
+#' @examples
+#' \dontrun{
+#' state_ <- "Georgia"
+#'
+#'# get state level covid data
+#' st_gen_df <- get_genstate_covid() %>%
+#'     filter(State == state_)
```
Comment (erika-tyagi) [inline R/get_genstate_covid.R:16]: Added this to the vaccine PR but should've noticed here too – could we add just `get_genstate_covid()` as an example that gets run? Extremely likely nothing breaks, but I like the idea of the CI catching if a link or something changes.

Comment (nmmarquez) [reply R/get_genstate_covid.R:16]: added!

File: R/get_genstate_covid.R (hunk 1 above)
Comment (nmmarquez) [reply R/get_genstate_covid.R:81]: good catch

**Review by erika-tyagi — APPROVED** [review]

**Pull request merged**